features/mbedtls/platform/mbed_trng.c

```diff
@@ -0,0 +1,29 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hal/trng_api.h"
+
+#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT) && defined(DEVICE_TRNG)
```

Comment:
It is enough to check for DEVICE_TRNG here, checking MBEDTLS_ENTROPY_HARDWARE_ALT is not necessary here, because it will be automatically defined whenever DEVICE_TRNG is present.

(Also since `mbedtls/config.h` is not included here, MBEDTLS_ENTROPY_HARDWARE_ALT would only be defined only if it is set in command line or in some application specific .json file)